### ISSUE TITLE
How to synchronize

### ISSUE THREAD

**Issue by LovelyBuggies** [henryiii/mini_hist#7]
Using this repository as a template, there is not upstream repo shown in `git remote -v`, and I cannot use `fetch` + `rebase` to renew my personal repo timely (nor can I use `git pull -rebase mini_hist master` + `git push origin master`). How could I synchronize this repo's renewal? Thanks!

**Comment by henryiii** [comment]
Hmm, that's an excellent question. I'll play with it and see - you might be able to add this as an upstream manually, and then sync. I'll try it tomorrow or at the latest, Monday.

**Comment by LovelyBuggies** [comment]
Yes, I thought about it yesterday, but I'm still trying. Looking forward to your reply. Thanks!

**Comment by henryiii** [comment]
This is an interesting question, but there's only been minor change(s) to the actual code, so you can simply implement them and move one - I'll try to find the answer to this question soon, though.

**Comment by henryiii** [comment]
Due to unrelated histories, it is not all that elegant, but you can do slightly better than copy and paste:

```bash
# Not really a related remote, but still valid to do in git
git remote add upstream git@github.com:henryiii/mini_hist
git fetch upstream

# Set everything to the upstream versions
git checkout upstream/master -- *

# Let's clear the staging area so we can pick what we want to keep
git reset
```

This is best if you left master as it was and all new work was on a feature branch instead. You can select what you want to keep with `git diff`, then `git add` what you want to keep (or use a GUI). You can then rebase or merge your feature branch with master.

**Comment by henryiii** [comment]
Note: in real work, we will be working on branches and forks, not templates. ;)

**Comment by henryiii** [comment]
This looks interesting, but not worth it in this case: https://github.com/rioam2/git-upstream-template
